community/kernel/src/main/java/org/neo4j/kernel/impl/api/OperationsFacade.java

```diff
@@ -439,6 +451,10 @@ public PrimitiveIntIterator relationshipGetPropertyKeys( long relationshipId ) t
         {
             return relationship.get().getPropertyKeys();
         }
+        finally
+        {
+            statement.assertOpen();
+        }
```

Comment:
Why do we have this check at the end? Even if the iterator is lazy, this check will come too soon to protect anything, won't it?